akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -805,6 +811,43 @@ abstract class ShardCoordinator(
           case _              => //Reallocated to another region
         }
 
+      case StopShards(shardIds) =>
+        if (settings.rememberEntities) {
+          log.error(
+            "{}: Stop shards cannot be combined with remember entities, ignoring command to stop [{}]",
+            typeName,
+            shardIds)
+        } else if (state.regions.nonEmpty && !preparingForShutdown) {
+          waitingForRebalanceAcks = shardIds.foldLeft(waitingForRebalanceAcks) { (acc, shardId) =>
+            if (!state.shards.contains(shardId)) {
+              // shard not running, so ack right away
+              sender() ! ShardStopped(shardId)
+              acc
+            } else {
+              acc.get(shardId) match {
+                case Some(refs) => acc.updated(shardId, refs + sender())
+                case None       => acc.updated(shardId, Set(sender()))
+              }
+            }
+          }
+
+          // except not started and not already rebalancing
+          val shardsToStop =
+            shardIds.filter(shard => state.shards.contains(shard) && !rebalanceInProgress.contains(shard))
+          log.info("{}: Explicitly stopping shards [{}]", typeName, shardsToStop)
+
+          val shardsPerRegion =
+            shardsToStop.flatMap(shardId => state.shards.get(shardId).map(region => region -> shardId)).groupBy(_._1)
+          shardsPerRegion.foreach {
+            case (region, shards) => shutdownShards(region, shards.map(_._2))
+          }
+        } else {
+          log.warning(
+            "{}: Explicit stop shards of shards [{}] ignored (no known regions or sharding shutting down)",
+            typeName,
+            shardIds)
+        }
```

Comment:
no need to reply with `ShardStopped` here?

Reply:
I was thinking of it as a failure, but maybe seeeing it as all stopped is more reasonable.

Reply:
I was mostly copy-pasting from the rebalance tick, if shutting down the cluster ack could make sense, but not important because we are anyway shutting down, if the coordinator does not know any regions yet it's better for the requesting side to time out and retry once coordinator has seen regions?